### ISSUE TITLE
hostname based profiles

### ISSUE THREAD

**Issue by johrstrom** [OSC/ondemand#2127]
With profile support, I'd like to have automatic profile support based on the hostname being requested. That is, we **force** the profile given the hostname. It's not a choice folks have, it's an administrative way to create many portals and simply have a few DNS names.

I'm thinking the name of the host is the name of the profile

```yaml
profiles:
  ondemand.osc.edu:
     # ...
  class.osc.edu:
    # ...
```

or we simply have a `hostname` in the structure.

```yaml
profiles:
  ondemnd:
    hostname: ondemand.osc.edu
  classroom:
    hostname: class.osc.edu
```

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1202671454765468) by [Unito](https://www.unito.io)

**Comment by adaybujeda** [comment]
I have an idea for an implementation based on using domains as profile names, eg:
```yaml
profiles:
  ondemand.osc.edu:
     # ...
  class.osc.edu:
    # ...
```

Draft PR: https://github.com/OSC/ondemand/pull/2173

**Comment by adaybujeda** [comment]
Small demo of the changes in the PR:

![domain](https://user-images.githubusercontent.com/17336210/181491017-6a3d5042-810d-4347-b256-0b259f16a776.gif)

**Comment by adaybujeda** [comment]
Showed the demo to the IQSS team and Bill suggested that having the configuration within the profile itself, as in the second proposal might be the better approach as it allows to configure multiple domains for the same profile.

```yaml
profiles:
  ondemnd:
    hostnames: ["ondemand.osc.edu", "alias1.osc.edu", "alias2.osc.edu"]
  classroom:
    hostnames: ["class.osc.edu"]
```

**Comment by adaybujeda** [comment]
Or similar to VHosts in Apache, add an `alias` property for the name:

```yaml
profiles:
  ondemand:
    alias: ["other_profile_name", "ondemand.osc.edu", "domain1.osc.edu"]
  classroom:
    alias: ["class", "class.osc.edu"]
```

**Comment by johrstrom** [comment]
This was fixes in #2173
